### DISCUSSION TITLE
Add support for BIN books

### DISCUSSION THREAD

**Discussion by rubenkku** [official-stockfish/Stockfish#5119] [General]
Hello Stockfish community, i would like to make a pourpose:
- Can you add support for BIN (polyglot) books for Stockfish?
Just a UCI setting where you (optionally) can specify a path for BIN book, so Stockfish can move instantly from the moves it reads on the BIN book.
Some GUI dont have BIN support, and adding this feature would make it usable under all existing GUI
BIN format is opensource, and it would not be complicated to add support for it.
Many thanks

**Comment by PGG106** [comment]
this was asked recently and a fork was made by someone to support books (for now), you can probably deduce by the fact polyglot books aren't supported and haven't been supported for years that the stockfish project doesn't really care about them.

**Reply by PGG106** [reply]
this is the previous discussion about this: https://github.com/official-stockfish/Stockfish/discussions/5081

**Reply by rubenkku** [reply]
Yes, i read the other post. Thanks pointing at it.

But what i ask here is just incorporate BIN book support by default to all next Stockfish updates, just like an aditional feature.

**Reply by PGG106** [reply]
that has a very close to 0 chance of happening based on historical precedents

**Reply by rubenkku** [reply]
ok so if you wont do it, then i did!!

https://github.com/rubenkku/SF-dev-XXXX-ChatGPT

**Comment by rubenkku** [comment]
Hi,
I wanted to share a project that might be interesting for developers and advanced users working with BIN opening books.
BIN (Polyglot) books are still widely used by computer chess enthusiasts, engine tournaments, and book editors, but Stockfish does not support them natively.
Because of that, I’ve been working on a small tool that provides:
A GUI application to view, edit, create, merge, and convert BIN books.
A source-code patcher that automatically modifies the Stockfish source to add native BIN book support, keeping the rest of the engine untouched.
It is not intended to change Stockfish behavior by default, but to offer an optional workflow for those who still rely on BIN books in their setups.

https://github.com/rubenkku/PGN-Book-Editor/releases
